test/SILOptimizer/no_external_def_to_decl_for_transparent.swift

```diff
@@ -0,0 +1,30 @@
+// RUN: rm -rf %t && mkdir -p %t
+// RUN: %target-build-swift %s -O -wmo -Xllvm -sil-disable-pass="Speculative Devirtualization" -o %t/a.out
+// RUN: %target-swift-frontend %s -O -Xllvm -sil-disable-pass="Speculative Devirtualization" -emit-ir | %FileCheck %s
+
+// Check if we generate code for the Int.== transparent function and don't get an unresolved symbol error for it.
+
+// CHECK-LABEL: define void @{{.*}}testit{{.*}} {
+// CHECK: store {{.*}} @_TFsoi2eeFTSiSi_Sb  
+// CHECK: ret void
+
+// CHECK-LABEL: define linkonce_odr hidden i1 @_TFsoi2eeFTSiSi_Sb(i64, i64) {{.*}} {
+
+var gg : ((Int, Int) -> Bool)?
+
+public class X {
+
+  func testit() {
+    gg = (==)
```

Comment:
For this test to be meaningful, the transparent function must come from another module. I didn't want to create my own module for this, because it would complicate the test (I would need to add another file).
AFIAK, the transparency of == is not an implementation detail, it's ABI.
I'd prefer to keep it as is.